packages/flutter_driver/test/src/real_tests/timeline_test.dart

```diff
@@ -56,5 +56,53 @@ void main() {
       expect(e2.threadTimestampMicros, isNull);
       expect(e2.arguments, isNull);
     });
+
+    test('sorts JSON', () {
+      final Timeline timeline = Timeline.fromJson(<String, dynamic>{
+        'traceEvents': <Map<String, dynamic>>[
+          <String, dynamic>{
+            'name': 'test event 1',
+            'ts': 456,
+          },
+          <String, dynamic>{
+            'name': 'test event 2',
+            'ts': 457,
+          },
+        ],
+      });
+
+      expect(timeline.events, hasLength(2));
+      expect(timeline.events[0].timestampMicros, equals(456));
+      expect(timeline.events[1].timestampMicros, equals(457));
+    });
+
+    test('sorts JSON nulls first', () {
+      final Timeline timeline = Timeline.fromJson(<String, dynamic>{
+        'traceEvents': <Map<String, dynamic>>[
+          <String, dynamic>{
+            'name': 'test event 0',
+            'ts': null,
+          },
+          <String, dynamic>{
+            'name': 'test event 1',
+            'ts': 456,
+          },
+          <String, dynamic>{
+            'name': 'test event 2',
+            'ts': 457,
+          },
+          <String, dynamic>{
+            'name': 'test event 3',
+            'ts': null,
+          },
+        ],
+      });
+
+      expect(timeline.events, hasLength(4));
+      expect(timeline.events[0].timestampMicros, isNull);
```

Comment:
I would throw in an expect for the event names as well just for completeness. It may seem silly, but this doesn't technically test that the events themselves are reordered when sorting the timestamps...